src/sysnet/supdup.325

```diff
@@ -273,9 +281,24 @@ SUPDU1:	MOVEI A,HSTPAG
 	 .VALUE
 	MOVEI A,JCLBUF
 	PUSHJ P,NETWRK"HSTLOOK	;GET HOST NUMBER INTO A, NETWORK NUMBER INTO TT
+ifn DODNS,[
+	 ;; NETWRK failed, try DNS
```

Comment:
Why can't NETWRK"HSTLOOK do DNS under the covers?

Reply:
It would make sense, except it would (under the hood) make many programs larger and more complex/slower. Consider e.g. PEEK which is in a sense a real-time program, and might need to resolve addresses quite a few times per time unit in some circumstances. 

The other way around might be more reasonable: let RESOLV use NETWRK under the hood? 

But it's really unsustainable to have "all these programs" do both, in a manual fashion, reproducing the same code.

Reply:
I see your point about size/speed considerations.  I don't know what PEEK does with hostnames.  Would it make sense for programs to say $$DNS==0 or 1 before .INSRTing NETWRK to control whether to use DNS or not?  It seems to me HSTLOOK is the natural entry point for any program that want to look up a host name, and it might be possible to extend this API in a way to make all callers happy.

Reply:
And then again, NETWRK is not only about name/address resolution, and RESOLV is not only about name<->address but also other DNS things. Hmm.

Reply:
Ho hum, reading the code is a way to learn. DQDEV already uses NETWRK if asked for it (e.g. domain:HOSTS3;ch;a;no.nocrew.org rather than domain:QUERY;ch;a;no.nocrew.org), so making RESOLV first ask for HOSTS3 and then DNS data should be easy - not efficient, but easy. ;-)